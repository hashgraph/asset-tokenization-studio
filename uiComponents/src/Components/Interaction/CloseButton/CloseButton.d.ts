/// <reference types="react" />
import type { IconButtonProps } from "@/Components/Interaction/IconButton";
export interface CloseButtonProps extends Omit<IconButtonProps, "aria-label"> {
}
export declare const CloseButton: ({ ...props }: CloseButtonProps) => JSX.Element;
