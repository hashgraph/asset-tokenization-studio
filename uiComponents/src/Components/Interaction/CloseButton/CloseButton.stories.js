var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import React from "react";
import { CloseButton } from "./CloseButton";
import { iconButtonSizes } from "@/storiesUtils";
var meta = {
    title: "Design System/Interaction/CloseButton",
    component: CloseButton,
    argTypes: {
        size: {
            options: iconButtonSizes,
            control: { type: "radio" }
        },
        isDisabled: {
            control: { type: "boolean" }
        }
    },
    args: {
        size: "md"
    },
    parameters: {
        design: {
            type: "figma",
            url: "https://www.figma.com/file/yn0TWBlB8x5WbsEY8BdML7/Palm-Design-System?node-id=1353%3A10670"
        },
        docs: {}
    }
};
export default meta;
var Template = function (args) { return (React.createElement(CloseButton, __assign({}, args))); };
export var Small = Template.bind({});
Small.args = {
    size: "sm"
};
export var Medium = Template.bind({});
Medium.args = {
    size: "md"
};
export var Large = Template.bind({});
Large.args = {
    size: "lg"
};
export var IsDisabled = Template.bind({});
IsDisabled.args = {
    size: "lg",
    isDisabled: true
};
