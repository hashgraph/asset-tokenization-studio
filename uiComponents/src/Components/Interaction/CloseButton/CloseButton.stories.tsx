import type { Meta, StoryFn } from "@storybook/react";
import React from "react";
import { CloseButton } from "./CloseButton";
import { iconButtonSizes } from "@/storiesUtils";

const meta = {
  title: "Design System/Interaction/CloseButton",
  component: CloseButton,
  argTypes: {
    size: {
      options: iconButtonSizes,
      control: { type: "radio" },
    },
    isDisabled: {
      control: { type: "boolean" },
    },
  },
  args: {
    size: "md",
  },
  parameters: {
    design: {
      type: "figma",
      url: "https://www.figma.com/file/yn0TWBlB8x5WbsEY8BdML7/Palm-Design-System?node-id=1353%3A10670",
    },
    docs: {},
  },
} as Meta<typeof CloseButton>;
export default meta;

const Template: StoryFn<typeof CloseButton> = (args) => (
  <CloseButton {...args} />
);

export const Small = Template.bind({});
Small.args = {
  size: "sm",
};

export const Medium = Template.bind({});
Medium.args = {
  size: "md",
};

export const Large = Template.bind({});
Large.args = {
  size: "lg",
};

export const IsDisabled = Template.bind({});
IsDisabled.args = {
  size: "lg",
  isDisabled: true,
};
