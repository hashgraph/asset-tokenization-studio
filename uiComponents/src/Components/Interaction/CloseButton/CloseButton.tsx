import React from "react";
import type { IconButtonProps } from "@/Components/Interaction/IconButton";
import { IconButton } from "@/Components/Interaction/IconButton";
import { X } from "@phosphor-icons/react";
import { PhosphorIcon } from "@/Components/Foundations";

export interface CloseButtonProps extends Omit<IconButtonProps, "aria-label"> {}

export const CloseButton = ({ ...props }: CloseButtonProps) => {
  return (
    <IconButton
      aria-label="close-button"
      icon={<PhosphorIcon as={X} />}
      variant={"tertiary"}
      {...props}
    />
  );
};
