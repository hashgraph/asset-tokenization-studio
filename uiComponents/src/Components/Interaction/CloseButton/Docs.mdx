import {
  Story,
  Meta,
  Title,
  ArgTypes,
  Primary,
  Stories,
  Source,
} from "@storybook/addon-docs";
import { Grid, GridItem, Text, Link, Center, Flex } from "@chakra-ui/react";
import * as CloseButtonStories from "./CloseButton.stories";

<Meta of={CloseButtonStories} />

<Title />

<a
  href="https://www.figma.com/file/YF8AvIq0MclfhGDvKgzypp/ioBricks-Design-System?type=design&node-id=1503%3A23013&mode=design&t=KlerdyEMWSQWLAVX-1"
  target="\_blank"
>
  View figma design
</a>

<Primary />

## How to set up

CloseButton is built with IconButton component.

## Props

<ArgTypes of={CloseButtonStories} />

## Stories

<Stories title="" />
