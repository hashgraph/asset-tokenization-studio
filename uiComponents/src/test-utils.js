import { ChakraProvider, extendTheme } from "@chakra-ui/react";
import { render as testingLibraryRender } from "@testing-library/react";
import React from "react";
import { BasePlatformTheme } from "./Theme/theme";
var DefaultTheme = extendTheme(BasePlatformTheme);
export var render = function (ui, theme) {
    if (theme === void 0) { theme = DefaultTheme; }
    var wrapper = function (_a) {
        var children = _a.children;
        return (React.createElement(ChakraProvider, { theme: theme }, children));
    };
    return testingLibraryRender(ui, { wrapper: wrapper });
};
