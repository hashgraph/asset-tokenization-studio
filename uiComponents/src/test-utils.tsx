import { ChakraProvider, extendTheme } from "@chakra-ui/react";
import type { RenderResult } from "@testing-library/react";
import { render as testingLibraryRender } from "@testing-library/react";
import React from "react";
import { BasePlatformTheme } from "./Theme/theme";

const DefaultTheme = extendTheme(BasePlatformTheme);

export const render = (ui: JSX.Element, theme = DefaultTheme): RenderResult => {
  const wrapper = ({ children }: { children?: React.ReactNode }) => (
    <ChakraProvider theme={theme}>{children}</ChakraProvider>
  );

  return testingLibraryRender(ui, { wrapper });
};
