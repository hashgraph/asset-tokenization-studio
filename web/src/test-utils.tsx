import React from "react";
import { ChakraProvider } from "@chakra-ui/react";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import type { RenderOptions, RenderResult } from "@testing-library/react";
import { render, waitFor } from "@testing-library/react";
import type { MemoryHistory } from "history";
import { createMemoryHistory } from "history";
import { I18nextProvider } from "react-i18next";
import { Router } from "react-router-dom";
import i18n from "./i18n";
import theme from "./theme";
import userEvent from "@testing-library/user-event";

export const selectCalendar = async (
  component: RenderResult,
  id: string,
  day: string | number = new Date().getDate(),
) => {
  const calendar = component.getByTestId(id);
  await userEvent.click(calendar);

  await waitFor(() => {
    const daysToSelect = component.getAllByTestId(`day-${day}`);

    const dayToSelect = daysToSelect.find(
      (day) => !day.hasAttribute("disabled"),
    );

    if (dayToSelect) {
      userEvent.click(dayToSelect);
    }
  });

  await userEvent.click(document.body);
};

export const testQueryClient = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
      refetchOnMount: true,
      cacheTime: 0,
    },
  },
  logger: {
    log: console.log,
    warn: console.warn,
    // âœ… no more errors on the console for tests
    error: () => {},
  },
});

export const queryWrapper = ({ children }: { children: React.ReactNode }) => (
  <QueryClientProvider client={testQueryClient}>{children}</QueryClientProvider>
);

const memoryHistory = createMemoryHistory();

export const AllProviders = ({
  children,
  history = memoryHistory,
}: {
  children?: React.ReactNode;
  history?: MemoryHistory;
}) => {
  return (
    <I18nextProvider i18n={i18n}>
      <QueryClientProvider client={testQueryClient}>
        <ChakraProvider theme={theme}>
          <Router location={history.location} navigator={history}>
            {children}
          </Router>
        </ChakraProvider>
      </QueryClientProvider>
    </I18nextProvider>
  );
};

const customRender = (
  ui: React.ReactElement,
  {
    options,
    history,
  }: {
    options?: RenderOptions;
    history?: MemoryHistory;
  } = {},
): RenderResult =>
  render(ui, {
    wrapper: ({ children }) => (
      <AllProviders history={history}>{children}</AllProviders>
    ),
    ...options,
  });

export { customRender as render };
