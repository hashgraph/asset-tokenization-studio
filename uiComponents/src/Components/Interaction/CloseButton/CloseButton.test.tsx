import { render } from "@/test-utils";
import { waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import React from "react";
import type { CloseButtonProps } from "./CloseButton";
import { CloseButton } from "./CloseButton";

describe(`<CloseButton />`, () => {
  const defaultProps: CloseButtonProps = {
    onClick: jest.fn(),
  };

  const baseSizesList = ["sm", "md", "lg"];
  const factoryComponent = (props: CloseButtonProps = defaultProps) =>
    render(<CloseButton {...props} />);

  test("renders correctly", () => {
    const component = factoryComponent();

    expect(component.asFragment()).toMatchSnapshot();
  });

  baseSizesList.forEach((size) => {
    test(`renders correctly with size ${size}`, () => {
      const component = factoryComponent({
        ...defaultProps,
        size,
      });

      expect(component.asFragment()).toMatchSnapshot(`CloseButton-${size}`);
    });
  });

  test("should pass down props", async () => {
    const component = factoryComponent();

    await userEvent.click(component.getByLabelText("close-button"));

    await waitFor(() => {
      expect(defaultProps.onClick).toHaveBeenCalled();
    });
  });
});
