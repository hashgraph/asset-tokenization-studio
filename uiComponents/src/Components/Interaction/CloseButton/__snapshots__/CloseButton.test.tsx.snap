// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CloseButton /> renders correctly 1`] = `
<DocumentFragment>
  <button
    aria-label="close-button"
    class="chakra-button css-ejr22g"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="chakra-icon css-kpbtv0"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="0 0 256 256"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M205.66,194.34a8,8,0,0,1-11.32,11.32L128,139.31,61.66,205.66a8,8,0,0,1-11.32-11.32L116.69,128,50.34,61.66A8,8,0,0,1,61.66,50.34L128,116.69l66.34-66.35a8,8,0,0,1,11.32,11.32L139.31,128Z"
      />
    </svg>
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<CloseButton /> renders correctly with size lg: CloseButton-lg 1`] = `
<DocumentFragment>
  <button
    aria-label="close-button"
    class="chakra-button css-6whudu"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="chakra-icon css-kpbtv0"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="0 0 256 256"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M205.66,194.34a8,8,0,0,1-11.32,11.32L128,139.31,61.66,205.66a8,8,0,0,1-11.32-11.32L116.69,128,50.34,61.66A8,8,0,0,1,61.66,50.34L128,116.69l66.34-66.35a8,8,0,0,1,11.32,11.32L139.31,128Z"
      />
    </svg>
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<CloseButton /> renders correctly with size md: CloseButton-md 1`] = `
<DocumentFragment>
  <button
    aria-label="close-button"
    class="chakra-button css-ejr22g"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="chakra-icon css-kpbtv0"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="0 0 256 256"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M205.66,194.34a8,8,0,0,1-11.32,11.32L128,139.31,61.66,205.66a8,8,0,0,1-11.32-11.32L116.69,128,50.34,61.66A8,8,0,0,1,61.66,50.34L128,116.69l66.34-66.35a8,8,0,0,1,11.32,11.32L139.31,128Z"
      />
    </svg>
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<CloseButton /> renders correctly with size sm: CloseButton-sm 1`] = `
<DocumentFragment>
  <button
    aria-label="close-button"
    class="chakra-button css-rlzu75"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="chakra-icon css-kpbtv0"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="0 0 256 256"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M205.66,194.34a8,8,0,0,1-11.32,11.32L128,139.31,61.66,205.66a8,8,0,0,1-11.32-11.32L116.69,128,50.34,61.66A8,8,0,0,1,61.66,50.34L128,116.69l66.34-66.35a8,8,0,0,1,11.32,11.32L139.31,128Z"
      />
    </svg>
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
